nefertari/events.py

```diff
@@ -0,0 +1,198 @@
+from contextlib import contextmanager
+
+from zope.interface import (
+    Attribute,
+    Interface,
+)
+from zope.interface import implementer
+
+
+class IRequestEvent(Interface):
```

Comment:
I'm not sure why you are using an interface here. You don't need to use interfaces for events. It's only really helpful if you don't want to be tied to your concrete classes. But in this case I don't see the advantage, though maybe I haven't read far enough into the PR to see it yet.

In any event, it's fine to have subscribers for RequestEvent, rather than for IRequestEvent.